core/src/swarm/swarm.rs

```diff
@@ -258,7 +273,11 @@ where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
                     self.behaviour.inject_node_event(conn_info.peer_id().clone(), event);
                 },
                 Async::Ready(RawSwarmEvent::Connected { conn_info, endpoint }) => {
-                    self.behaviour.inject_connected(conn_info.peer_id().clone(), endpoint);
+                    if self.banned_peers.contains(conn_info.peer_id()) {
+                        self.raw_swarm.peer(conn_info.peer_id().clone()).into_connected().map(|c| c.close());
```

Comment:
I actually preferred the `expect` here. If `into_connected` returns `None` then there's a very serious bug and it's preferable to panic rather than silently continue in an invalid state.